File: 프로그래머스/lv2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def solution(n, t, m, p):
    answer = ''
    nums = '0'
    base = '0123456789ABCDEF'
    for k in range(0,m*t):
        temp = []
        while k>0:
            temp.append(base[k%n])
            k //= n
        temp.reverse()
        nums += "".join(temp)
    for i in range(t):answer += nums[m*i+p-1]
    return answer

File: 프로그래머스/lv2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    int n;
    
    String nBase(int num) {
        String result = "";
        String[] base = "ABCDEF".split("");
        int rest;
        
        if (num == 0) {
            return "0";
        }
        
        while (num > 0) {
            rest = num % n;
            if (rest > 9) {
                result = base[rest % 10] + result;
            } else {
                result = String.valueOf(rest) + result;
            }
            num = num / n;
        }
        return result;
    }
    
    public String solution(int n, int t, int m, int p) {
        String answer = "";
        String s = "";
        int num = 0;
        int temp = 0;
        this.n = n;
        
        while (s.length() < t*m) {
            s += nBase(num++);
        }
        
        for (int i=0;i<s.length();i++) {
            if (i % m + 1 == p) {
                temp++;
                answer += s.substring(i,i+1);
            }
            if (temp == t) {
                break;
            }
        }
        
        System.out.println(s);

        return answer;
    }
}